### PR TITLE
Chore: setup simple tests and github actions

### PR DESCRIPTION
**Pull request by pdpino**
* Simple tests: `App` and `WeekView` render correctly
* Test workflows:
  * `test-main`: test library source code
  * (removed) ~~`test-example`: test example App code by itself~~
  * (removed) ~~`test-example-in-local`: test example App as it would be run by a developer (to avoid issues like this: https://github.com/hoangnm/react-native-week-view/pull/114#discussion_r680345968)~~
* Add simple developing instructions
* Setup yarn scripts for installation in example App (see details in next comment)

ps: I will setup `linter.yml` later, when other PRs are closed (to avoid code conflicts)

### CODE REVIEW

**Comment by pdpino** [comment]
Yarn example/ scrips details:
* react-native/metro do not work with symlinks or files (e.g. `"react-native-week-view": "file://.."`) --> break the app and/or the tests
* Solution:
```sh
# 1. install everything
cd example
yarn install

# 2. replace node_modules/react-native-week-view/ using rsync
yarn install:local
```

* The following scripts are used:
  * `install:buildlink`: Builds a link for the week-view project, in case it does not exist
  * `install:rmlink`: links and then removes the link in `node_modules/react-native-week-view/` (this is much faster than deleting the whole folder with `rm -rf`)
  * `install:sync`: copies the relevant files from week-view to `node_modules/react-native-week-view/` with rsync
* Some wrappers, for ease of use:
  * `install:rmfile`: calls buildlink + rmlink (--> removes the folder in `node_modules/react-native-week-view/`)
  * `install:local`: calls rmfile + sync (--> replaces the yarn installation by the rsync installation)

**Comment by hoangnm** [comment]
Hi @pdpino , thanks for the PR, it looks promising. Could you help to explain again why we need 3 test actions? I'm a bit confused here.

**Comment by pdpino** [comment]
Sure!

1. `test-main`: tests the library by itself (ignore the example app completely)
2. `test-example`: tests the example app with jest
3. `test-example-in-local`: tests the example app as would be run by a developer (e.g. us) in their machines.

Before (https://github.com/hoangnm/react-native-week-view/pull/114#discussion_r680345968) we had the following issue: the independent tests run properly (1 and 2), but we could not run the app in the local machine, throwing this error:
```
bundling failed: ambiguous resolution: module `/.../react-native-week-view/example/index.js` tries to require `react-native`, but there are several files providing this module. You can delete or fix them
```
The test 3 should catch this. React-native / metro do not work properly with installing libraries from links (`react-native-week-view: "file:../"`), so with the `install:local` we test that.

Hope this clarifies it

**Comment by pdpino** [comment]
@hoangnm now that I see it again, I think the test 2 could be removed. The example app is only run in our local machines for testing, so the test 3 should cover the test 2 as well.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v1
+      with:
+        node-version: 14.x
+
+    - run: yarn install --frozen-lockfile
+    - run: yarn test
+
+  test-example-in-local:
```
Comment (hoangnm) [inline .github/workflows/tests.yml:22]: Hi @pdpino , I'm wondering if we need this on the workflow?

Comment (pdpino) [reply .github/workflows/tests.yml:22]: @hoangnm you mean not testing the app by itself. Sounds good as well

Comment (hoangnm) [reply .github/workflows/tests.yml:22]: yeah, in my view, the flow should run the tests of the library only.

**Comment by hoangnm** [comment]
Thank @pdpino, I didn't know the tricks you did to make it works!

**Pull request merged**